### PR TITLE
feat(scm-github): Allow editing of existing PR bot comments

### PR DESCRIPTION
**Pull request by supra08**
## Context

Fixes screwdriver-cd/screwdriver#1714

## Objective

Before this, when multiple builds took place, it resulted in multiple meta PR comments. To tackle that, the comments for that PR is listed and if a bot comment is found, it is edited, else a new comment is created.

## References

screwdriver-cd/screwdriver#1714

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -236,6 +236,66 @@ class GithubScm extends Scm {
...
+                    repo: scmInfo.repo,
+                    comment_id: commentId,
+                    body: comment
+                }
+            });
+
+            return pullRequestComment;
+        } catch (err) {
+            logger.error('Failed to edit PR comment: ', err);
+            throw err;
```
Comment (tkyi) [inline index.js:295]: probably don't need to throw error, just return null

File: index.js [hunk 2]
```diff
@@ -236,6 +236,66 @@ class GithubScm extends Scm {
...
+        } catch (err) {
+            logger.error('Failed to fetch PR comments: ', err);
+
+            return null;
+        }
+    }
+
+    /**
+     * Edit a particular comment in the PR
+     * @async  editPrCpmment
```
Comment (tkyi) [inline index.js:272]:
```suggestion
     * @async  editPrComment
```

File: index.js (hunk 1 above)
Comment (supra08) [reply index.js]: done

File: index.js (hunk 2 above)
Comment (supra08) [reply index.js]: done

**Pull request merged**